File: src/main/java/com/skybayninehundredninetynine/service/ReservationAgentService.java
package com.skybayninehundredninetynine.service;

import java.util.List;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.domain.Specification;
import org.springframework.http.ResponseEntity;

import com.skybayninehundredninetynine.domain.ReservationAgent;
import com.skybayninehundredninetynine.dto.ReservationAgentDTO;
import com.skybayninehundredninetynine.dto.ReservationAgentSearchDTO;
import com.skybayninehundredninetynine.dto.ReservationAgentPageDTO;
import com.skybayninehundredninetynine.dto.ReservationAgentConvertCriteriaDTO;
import com.skybayninehundredninetynine.service.GenericService;
import com.skybayninehundredninetynine.dto.common.RequestDTO;
import com.skybayninehundredninetynine.dto.common.ResultDTO;
import java.util.List;
import java.util.Optional;





public interface ReservationAgentService extends GenericService<ReservationAgent, Integer> {

	List<ReservationAgent> findAll();

	ResultDTO addReservationAgent(ReservationAgentDTO reservationAgentDTO, RequestDTO requestDTO);

	ResultDTO updateReservationAgent(ReservationAgentDTO reservationAgentDTO, RequestDTO requestDTO);

    Page<ReservationAgent> getAllReservationAgents(Pageable pageable);

    Page<ReservationAgent> getAllReservationAgents(Specification<ReservationAgent> spec, Pageable pageable);

	ResponseEntity<ReservationAgentPageDTO> getReservationAgents(ReservationAgentSearchDTO reservationAgentSearchDTO);
	
	List<ReservationAgentDTO> convertReservationAgentsToReservationAgentDTOs(List<ReservationAgent> reservationAgents, ReservationAgentConvertCriteriaDTO convertCriteria);

	ReservationAgentDTO getReservationAgentDTOById(Integer reservationAgentId);







}

File: skybayninehundredninetynine_ui/src/services/ReservationAgentService.js
import http from "../http-common"; 

class ReservationAgentService {
  getAllReservationAgents(searchDTO) {
    console.log(searchDTO)
    return this.getRequest(`/reservationAgent/reservationAgents`, searchDTO);
  }

  get(reservationAgentId) {
    return this.getRequest(`/reservationAgent/${reservationAgentId}`, null);
  }

  findByField(matchData) {
    return this.getRequest(`/reservationAgent?field=${matchData}`, null);
  }

  addReservationAgent(data) {
    return http.post("/reservationAgent/addReservationAgent", data);
  }

  update(data) {
  	return http.post("/reservationAgent/updateReservationAgent", data);
  }
  
  uploadImage(data,reservationAgentId) {
  	return http.postForm("/reservationAgent/uploadImage/"+reservationAgentId, data);
  }




	postRequest(url, data) {
		return http.post(url, data);
      };

	getRequest(url, params) {
        return http.get(url, {
        	params: params
        });
    };

}

export default new ReservationAgentService();
